### PR TITLE
Introduce farva-monitor

### PR DESCRIPTION
**Pull request by bcwaldon**
This new component simply syncs a set of EC2 instances based on a set of provided tags into the backend list of a given ELB.

### CODE REVIEW

File: cmd/farva-monitor/main.go [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+
+	"github.com/bcwaldon/farva/pkg/monitor"
+)
+
+func main() {
+	fs := flag.NewFlagSet("farva-monitor", flag.ExitOnError)
+
+	cfg := monitor.DefaultConfig
+
+	fs.DurationVar(&cfg.RefreshInterval, "refresh-interval", cfg.RefreshInterval, "Attempt to sync AWS entities at this interval")
```
Comment (jacobstr) [inline cmd/farva-monitor/main.go]: This was a bit confusing to me: we're both assigning to cfg.RefreshInterval and using it as a default value. It's
equivalent to `a = input or a` so maybe it's idiomatic but as someone new to go it's kind of confusing. I think just using `monitor.DefaultConfig.RefreshInterval` would be more readable.

Comment (bcwaldon) [inline cmd/farva-monitor/main.go]: Will fix.

File: cmd/farva-monitor/main.go [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+func (f *KVSliceFlag) String() string {
+	pairs := []string{}
+	for k, v := range *f {
+		pairs = append(pairs, fmt.Sprintf("%s=%s", k, v))
+	}
+	return strings.Join(pairs, ",")
+}
+
+func (f *KVSliceFlag) Set(value string) error {
+	pairs := strings.Split(value, ",")
```
Comment (jacobstr) [inline cmd/farva-monitor/main.go]: I think I'd feel better if this stripped whitespace around `, ` and `= `.

File: pkg/monitor/aws.go [hunk 3]
```diff
@@ -0,0 +1,160 @@
...
+	}
+
+	resp, err := svc.DescribeInstances(&req)
+	if err != nil {
+		return nil, err
+	}
+
+	instances := []string{}
+
+	//TODO(bcwaldon): need to handle resp.NextToken
```
Comment (jacobstr) [inline pkg/monitor/aws.go]: I'm intuiting this to occur when the number of `Reservations` is larger than the page size. How many is that?

File: wercker.yml [hunk 4]
```diff
@@ -9,16 +9,16 @@ build:
...
+        code: cp bin/linux_amd64/* "$WERCKER_OUTPUT_DIR"
 
 deploy:
     box: nginx:1.10.0
     steps:
     - internal/docker-push:
         username: $QUAY_USERNAME
         password: $QUAY_PASSWORD
-        cmd: /pipeline/source/farva-gateway
+        working-dir: /pipeline/source/
```
Comment (jacobstr) [inline wercker.yml]: The docker container built by wercker previously ran `farva`. With `farva-monitor` in the same container, do we now launch the container with different cmds explicitly provided?

Comment (bcwaldon) [inline wercker.yml]: You must provide a 'command' directive in the pod now. The other two options I see are to ship a single binary that takes an arg (e.g. `farva monitor` or `farva gateway`) or to ship two different images. Which of the three do you prefer?

Comment (jacobstr) [inline wercker.yml]: Looks like this was changed back. I'm kind of curious what the intention was.

**Comment by jacobstr** [comment]
👍 Looks good. I think the whitespace thing is a minor nit that may lead to unexpected results for folks with different whitespace habits - if the AWS api doesn't do it's own stripping.

File: pkg/monitor/aws.go (hunk 3 above)
Comment (bcwaldon) [inline pkg/monitor/aws.go]: Well the minimum page size is 5, and the max 1000. I can't find documentation of what the server-side actually provides, though. Given that we aren't targeting >5 nodes in any deployment I think it's safe.

File: cmd/farva-monitor/main.go (hunk 2 above)
Comment (bcwaldon) [inline cmd/farva-monitor/main.go]: Sure. I will add tests for this in my follow-on PR.

Comment (bcwaldon) [inline cmd/farva-monitor/main.go]: https://github.com/bcwaldon/farva/pull/5

File: wercker.yml (hunk 4 above)
Comment (bcwaldon) [inline wercker.yml]: It wasn't changed back. Take a look at master and let me know if you still want to talk.

**Pull request merged**